src/bst.py

```diff
@@ -0,0 +1,193 @@
+"""Classes for binary search tree.
+
+Methods include:
+insert(self, val): Insert value into tree; if value already exists, ignore it.
+search(self, val): Return node containing that value, else None.
+size(self): Return number of nodes/vertices in tree, 0 if empty.
+depth(self): Return number of levels in tree. Tree with one value has depth of 0.
+contains(self, val): Return True if value is in tree, False if not.
+balance(self): Return a positive or negative integer representing tree's balance.
+    Trees that are higher on the left than the right should return a positive value;
+    trees that are higher on the right than the left should return a negative value;
+    an ideally-balanced tree should return 0.
+"""
+
+from queue import Queue
+
+
+class Node():
+    """Node object for the binary search tree."""
+
+    def __init__(self, value, left=None, right=None):
+        """Instantiate a node object."""
+        self.value = value
+        self.left = left
+        self.right = right
+
+
+class BinarySearchTree():
+    """Binary Search Tree.
+
+    .insert(): Add a value to the BST
+    """
+
+    def __init__(self):
+        """Initialize the BinarySearchTree Class."""
+        self._size = 0
+        self.root = None
+
+    def insert(self, value):
+        """Insert a value in to the binary search tree."""
+        if self.root is None:
+            self.root = Node(value)
+            self._size = 1
+            return
+        current_node = self.root
+        while True:
+            if value < current_node.value:
+                if current_node.left:
+                    current_node = current_node.left
+                else:
+                    current_node.left = Node(value)
+                    self._size += 1
+                    break
+            elif value > current_node.value:
+                if current_node.right:
+                    current_node = current_node.right
+                else:
+                    current_node.right = Node(value)
+                    self._size += 1
+                    break
+            else:
+                break
+
+    def search(self, value):
+        """Search the Binary Search Tree for a value, return node or none."""
+        current_node = self.root
+        while True:
+            if value == current_node.value:
+                return current_node
+            if value < current_node.value:
+                if current_node.left:
+                    current_node = current_node.left
+                    continue
```

Comment:
Are the continue statements necessary? What if you used elif instead of a series of ifs?